temporian/implementation/numpy_cc/operators/tick_calendar.cc

```diff
@@ -28,36 +32,61 @@ py::array_t<double> tick_calendar(
   // Ticks list
   std::vector<double> ticks;
 
-  // Date range
-  const auto start_t = static_cast<std::time_t>(std::floor(start_timestamp));
-  const auto end_t = static_cast<std::time_t>(std::floor(end_timestamp));
+  // Direction in wich to search
+  int step = forward ? 1 : -1;
+
+  // End time
+  const bool end_time_set = end_timestamp >= 0;
```

Comment:
It look like a negative end_timestamp is a magic value. What about using std::optional instead?
Also, should this code work with negative timestamps?

Reply:
yes! that's much better